src/mpyl/steps/models.py

```diff
@@ -92,35 +93,41 @@ class RunProperties:
     console: ConsoleProperties
     """Settings for the console output"""
     stages: list[Stage]
-    """Stage definitions"""
+    """All stage definitions"""
+    projects: set[Project]
+    """All projects"""
+    run_plan: dict[Stage, set[Project]]
+    """Stages and projects for this run"""
 
     @staticmethod
     def for_local_run(
         config: dict,
+        run_plan: dict[Stage, set[Project]],
         revision: str,
         branch: Optional[str],
-        tag: Optional[str],
-        stages: list[dict],
+        tag: Optional[str] = None,
+        stages: Optional[list[dict]] = None,
+        all_projects: Optional[set[Project]] = None,
     ):
         return RunProperties(
             details=RunContext("", "", "", "", "", None),
             target=Target.PULL_REQUEST,
             versioning=VersioningProperties(revision, branch, 123, tag),
             config=config,
             console=ConsoleProperties("INFO", True, 130),
-            stages=[Stage(stage["name"], stage["icon"]) for stage in stages],
+            run_plan=run_plan or {},
+            stages=[Stage(stage["name"], stage["icon"]) for stage in stages]
+            if stages
+            else [],
+            projects=all_projects or set(),
         )
 
-    def to_stage(self, stage_name: str) -> Stage:
-        stage_by_name = next(stage for stage in self.stages if stage.name == stage_name)
-        if stage_by_name:
-            return stage_by_name
-        raise ValueError(f"Stage {stage_name} not found")
-
     @staticmethod
     def from_configuration(
         run_properties: dict,
         config: dict,
+        run_plan: Optional[dict[Stage, set[Project]]] = None,
+        all_projects: Optional[set[Project]] = None,
```

Comment:
Same here. If you make it non optional and pass an empty dict or set as params, you wouldn't need the `run_plan or {}` part below